ui/page/dcrdex/dcrdex_page.go

```diff
@@ -125,14 +130,23 @@ func (pg *DEXPage) Layout(gtx C) D {
 		msg = values.String(values.StrDexMainnetNotReady)
 	} else if !hasMultipleWallets {
 		msg = values.String(values.StrMultipleAssetRequiredMsg)
-	} else if !pg.AssetsManager.DEXCInitialized() || pg.CurrentPage() == nil {
-		msg = values.String(values.StrDEXInitErrorMsg)
+	} else if !pg.AssetsManager.DEXCInitialized() || pg.CurrentPage() == nil { // dexc must have been reset.
+		pg.showSplashPage = true
+		if !pg.dexIsLoading {
+			pg.setPage()
+		}
 	}
 
 	if msg != "" {
 		return components.DisablePageWithOverlay(pg.Load, nil, gtx, msg, "", actionBtn)
 	}
 
+	if pg.showSplashPage || pg.dexIsLoading {
+		return pg.Theme.List(pg.splashPageContainer).Layout(gtx, 1, func(gtx C, i int) D {
+			return pg.splashPage(gtx)
+		})
+	}
```

Comment:
I think return this to the top of this method. With this change, mainnet will never see the splash page and the user will also not see the splash page if he doesn't have multiple wallets. The splash page should display before any other page.

Reply:
> With this change, mainnet will never see the splash page and the user will also not see the splash page if he doesn't have multiple wallets.

Yes.

> The splash page should display before any other page.

Okay.